pkg/controller/controller.go

```diff
@@ -210,6 +225,48 @@ func createWorker(queue workqueue.RateLimitingInterface, resourceType string, ma
 	}()
 }
 
+func (c *controller) monitorConfigMap() {
+	// Cannot wait for the informer to push something into a queue.
+	// What we're waiting on may never exist without us configuring
+	// it, so we have to poll/ ask for it the first time to get it set.
+
+	// Can we ask 'through' an informer? Is it a writeback cache? I
+	// only ever want to monitor and be notified about one configmap
+	// in a hardcoded place.
+	glog.V(9).Info("cluster ID monitor loop enter")
+	cm, err := c.kubeClient.CoreV1().ConfigMaps("default").Get("cluster-info", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		m := make(map[string]string)
+		m["id"] = c.getClusterID()
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cluster-info",
+			},
+			Data: m,
+		}
+		// if we fail to set the id,
+		// it could be due to permissions
+		// or due to being already set while we were trying
+		if _, err := c.kubeClient.CoreV1().ConfigMaps("default").Create(cm); err != nil {
+			glog.Warningf("due to error %q, could not set clusterid configmap to %#v ", err, cm)
+		}
+	} else if nil == err {
+		// cluster id exists and is set
+		// get id out of cm
+		if id := cm.Data["id"]; "" != id {
+			c.setClusterID(id)
+		} else {
+			m := make(map[string]string)
+			cm.Data = m
+			m["id"] = c.getClusterID()
+			c.kubeClient.CoreV1().ConfigMaps("default").Update(cm)
```

Comment:
If the controller were the only component to read this value then allowing it to be in an admin-defined placed would seem ok. However, in the workflow where the Broker does call back to K8s and includes the clusterID, the component that processes the request from the Broker will need to read this ConfigMap too, to find the cluster with that ID. While its possible to require that other component to also have CLI flags and then require the admins who deploy svc-cat and this component to both use the same values, I wonder if its asking for too much synchronization between admins.

I'm not against giving them the flexibility to do so (I don't think anyway, in general I like to give people options), but I do think we should default to a well-known location so that the default/minimal install work out of the box and consider using a different 'name' or namespace to be an advanced option.

By "name", which name? Name of configMap? Name of the key? I assume you mean name of configMap.

Reply:
What's the pattern we're following for when something is a cmd line flag vs a config file property? Seems like something as permanent (unchanging) as where to store this info would be better suited for the config file. I tend to think of cmd line stuff as things that might need to be changed (or overwritten) more often than this would.

Reply:
> By "name", which name? Name of configMap? 

Yep, I want to be able to say "look in namespace X for configmap Y for the cluster ID". Having the name of the key always be "id" is totally fine.

> we should default to a well-known location

I'm fine with it defaulting to the current location (namespace "default" and "cluster-info").

Reply:
I don't want to give people the opportunity to hide this away or lose it. The goal is to be a standard, in a standard place where everyone can see it and coordinate.

Reply:
Is this a blocker or can we do it in a follow up?

Reply:
Let's go ahead and do it now. I think the flexibility might be needed in some environments where they are picky about where certain component's data lives.